math/factorial.cpp

```diff
@@ -5,10 +5,8 @@
 #include <iostream>
 
 /** function to find factorial of given number */
-unsigned int factorial(unsigned int n) {
-    if (n == 0)
-        return 1;
-    return n * factorial(n - 1);
+int64_t factorial(unsigned int n) {
```

Comment:
the suggested improvement wouldn't really happen until the input argument type is also explicitly specified.
```suggestion
int64_t factorial(uint64_t n) {
```

Reply:
Input can be int only because what expected is the output should be correct because output is going to be bigger number. So i think we don't need to change input type.

Reply:
You have a recursive function i.e., the output of the function in one instance is also the input in some next recursion. Hence, the input and the output **must** be of the same datatype. Your tests might be working because the compiler and the machine you are running on might have a default 64-bit or 32-bit integer that is satisfying the condition.